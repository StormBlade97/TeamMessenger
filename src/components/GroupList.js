import GroupItem from './GroupItem'
import React from 'react'

const GroupList = (props) => {

    const groupItems = props.groupList.map(groupItem => <GroupItem key={groupItem.id} name={groupItem.name} />);

    return (
        <ul>
            {groupItems}
        </ul>
    )
}

export default GroupList