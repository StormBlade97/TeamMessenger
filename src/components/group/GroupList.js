import React from 'react'
import GroupItem from './GroupItem'

const GroupList = (props) => {

    return (
        <ul>
            {props.groupList.map(groupItem =>
                <GroupItem
                    onClick={(groupId) => {
                        props.onGroupItemSelected(groupId)
                    }}
                    key={groupItem.id}
                    id={groupItem.id}
                    name={groupItem.name}/>
            )}
        </ul>
    )
}

export default GroupList