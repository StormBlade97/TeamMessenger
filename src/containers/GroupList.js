import GroupItem from '../components/GroupItem'
import React from 'react'
import {connect} from 'react-redux'
import {fetchGroupItems} from '../actions/GroupActions'


export class GroupList extends React.Component {

    constructor(props) {
        super(props)
    }

    componentDidMount() {
        const {dispatch} = this.props
        dispatch(fetchGroupItems())
    }

    render() {

        const groups = this.props.groups

        // Turn store.groups into map
        const groupList = Object.keys(groups).map((id) => <GroupItem key={id} name={groups[id].name}/>)

        return (
            <ul>
                {groupList}
            </ul>
        )
    }
}

const mapStatesToProps = state => ({
    groups: state.groups.byId
})

export default connect(mapStatesToProps)(GroupList)