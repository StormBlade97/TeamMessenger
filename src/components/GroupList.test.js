import React from 'react';
import GroupList from './GroupList';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

describe('Component: GroupList', () => {

    it('renders a snapshot', () => {

        const mockGroupList = [
            {name: 'cashiers', id: 1},
            {name: 'storage', id: 2}
        ]

        const tree = renderer.create(<GroupList groupList={mockGroupList}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})