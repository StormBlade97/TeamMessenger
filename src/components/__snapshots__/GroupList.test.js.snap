exports[`Component: GroupList renders a snapshot 1`] = `
<ul>
  <li>
    cashiers
  </li>
  <li>
    storage
  </li>
</ul>
`;
